core/frontend/src/BriefcaseConnection.ts

```diff
@@ -37,12 +54,17 @@ export interface GenericAbortSignal {
  * @public
  */
 export interface PullChangesOptions {
-  /** Function called regularly to report progress of changes download. */
+  /**
+   * Function called regularly to report progress of changes download.
+   * @deprecated Use [[downloadProgressCallback]] instead. Deprecated since 3.6
```

Comment:
thx for tagging when the deprecation occurred :)

Reply:
I was thinking: wouldn't it be more useful to tag when deprecated item will be removed ? That way user wouldn't have to look up our deprecation policy if he doesn't know it already

Reply:
User probably cares more about when it will be removed rather when it was deprecated.

Reply:
I'm going to defer this to @pmconne and @calebmshafer .

My 2cents:
I think it's ok to leave it as is. Just because an API has been deprecated doesnt mean it will ever be removed, although imo we should. It also might be removed past the original N+2 major we deprecated in (N) so don't want to necessarily give them any false expectations.

Reply:
Agree with @aruniverse. Policy only specifies minimum major release in which it might be removed. Also policy could conceivably change in future.

I do want to update guidelines to standardize the tag like "deprecated in [version]. [Suggestions for adjusting existing code]."
` * @deprecated in 3.6. Use [[downloadProgressCallback]] instead.`

Reply:
#4907.

Reply:
```suggestion
   * @deprecated in 3.6. Use [[downloadProgressCallback]] instead.
```